File: app/src/main/java/com/github/batulovandrey/unofficialurbandictionary/presenter/DetailPresenter.java
package com.github.batulovandrey.unofficialurbandictionary.presenter;

import com.github.batulovandrey.unofficialurbandictionary.bean.DefinitionResponse;

/**
 * @author Andrey Batulov on 29/10/2017
 */

public interface DetailPresenter {

    void setImageResToImageView(DefinitionResponse definition);

    DefinitionResponse getDefinitionByDefIf(long defId);

    boolean isFavoriteDefinition(DefinitionResponse definition);

    void isAddedToFav(long defId);
}

File: app/src/main/java/com/github/batulovandrey/unofficialurbandictionary/presenter/DetailPresenter.kt
package com.github.batulovandrey.unofficialurbandictionary.presenter

import com.github.batulovandrey.unofficialurbandictionary.bean.DefinitionResponse

/**
 * @author Andrey Batulov on 22/12/2017
 */

interface DetailPresenter {

    fun setImageResToImageView(definition: DefinitionResponse)

    fun getDefinitionByDefId(defId: Long): DefinitionResponse

    fun isFavoriteDefinition(definition: DefinitionResponse): Boolean

    fun isAddedToFav(defId: Long)
}